pyvista/utilities/misc.py

```diff
@@ -17,3 +18,123 @@ class PyvistaDeprecationWarning(Warning):
     """Non-supressed Depreciation Warning."""
 
     pass
+
+
+class VTKVersionInfo():
+    """Contains the VTK version information.
+
+    Wraps vtk.vtkVersion in a ``sys.version_info`` like manner.
+
+    Notes
+    -----
+    The micro attribute may be either the build date in YYYYMMDD or
+    the patch release.  Use caution when comparing version patch
+    release.
+
+    Examples
+    --------
+    Simply output the vtk version.
+
+    >>> import pyvista
+    >>> pyvista.vtk_version_info  # doctest:+SKIP
+    VTK Version 9.1.0
+
+    Compare the VTK version.
+
+    >>> pyvista.vtk_version_info > (9, 0)  # doctest:+SKIP
+    True
+
+    Return the major version of vtk.
+
+    >>> pyvista.vtk_version_info.major  # doctest:+SKIP
+    9
+    """
+
+    def __init__(self):
+        """Initialize the vtk version from the VTK class."""
+        # Get the VTK version. The API here is consistent since at least VTK v4.0.0
+        try:
+            ver = _vtk.vtkVersion()
+            self._major = ver.GetVTKMajorVersion()
+            self._minor = ver.GetVTKMinorVersion()
+            self._micro = ver.GetVTKBuildVersion()
+        except:  # pragma: no cover
+            self._major, self._minor, self._micro = (4, 0, 0)
+
+    @property
+    def major(self) -> int:
+        """Major vtk version.
+
+        Examples
+        --------
+        >>> import pyvista
+        >>> pyvista.vtk_version_info.major  # doctest:+SKIP
+        9
+        """
+        return self._major
+
+    @property
+    def minor(self) -> int:
+        """Minor vtk version.
+
+        Examples
+        --------
+        >>> import pyvista
+        >>> pyvista.vtk_version_info.minor  # doctest:+SKIP
+        1
+        """
+        return self._minor
+
+    @property
+    def micro(self) -> int:
+        """Micro vtk version.
+
+        This version may be either the released patch version or an
+        integer denoting the build date in YYYYMMDD.
+
+        Examples
+        --------
+        >>> import pyvista
+        >>> pyvista.vtk_version_info.micro  # doctest:+SKIP
+        0
+
+        """
+        return self._micro
+
+    def _as_tuple(self):
+        return (self.major, self.minor, self.micro)
+
+    def __gt__(self, other):
+        """Return tuple-like greater than."""
+        return self._as_tuple() > other
+
+    def __ge__(self, other):
+        """Return tuple-like greater than or equal to."""
+        return self._as_tuple() >= other
+
+    def __lt__(self, other):
+        """Return tuple-like less than than."""
+        return self._as_tuple() < other
+
+    def __le__(self, other):
+        """Return tuple-like less than or equal to."""
+        return self._as_tuple() <= other
+
+    def __eq__(self, other):
+        """Return tuple-like equal to."""
+        return self._as_tuple() == other
+
+    def __iter__(self):
+        """Yield tuple-like iterator."""
+        yield self._as_tuple()
+
+    def __contains__(self, other):
```

Comment:
I'm not sure what the use case of this is.  Based on the test, we can do `assert 9 in pyvista.vtk_version`.  This seems not useful to me as it checks whether any value is the same.

Reply:
I just wanted to make this like `sys.version_info`, where they're `tuple`-like. I agree that we don't need this, but this is mostly so we can make it look like the build-in python `version_info`.  I could go either way with this.

Reply:
I'm fine if this mirrors the python implementation.

Reply:
I don't think `__contains__` makes sense here, and the class is missing `__iter__` etc. so it's not a tuple anyway. I'd remove the needless bits. But I actually have a `namedtuple` recommendation in another comment.